src/m3ninx/index/segment/builder/terms.go

```diff
@@ -41,10 +42,12 @@ type termElem struct {
 }
 
 func newTerms(opts Options) *terms {
+	pool := opts.PostingsListPool()
 	return &terms{
-		opts:     opts,
-		pool:     opts.PostingsListPool(),
-		postings: NewPostingsMap(PostingsMapOptions{}),
+		opts:              opts,
+		pool:              pool,
+		postingsListUnion: pool.Get(),
```

Comment:
Think this needs to get reset in the call to `terms.reset()` here?

Maybe we should add a `TestTermsReuse(...)` that ensures it gets reset properly if we don't already have a reuse test?